### PR TITLE
feat(all): split core into graphql and state packages

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the new behavior?
- Moves `substream`, `DaffLoadingState`, and `DaffStoreFacade` into `@daffodil/core/state`.
- Moves `daffBuildFragmentDefinition` and `daffBuildFragmentNameSpread` into `@daffodil/core/graphql`.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

Some exports from `@daffodil/core` have been moved into `@daffodil/core/graphql` and `@daffodil/core/state`.

## Other information

### CODE REVIEW

File: libs/cart/README.md [hunk 1]
```diff
@@ -3,9 +3,66 @@
 Building and maintaining a model and code for an ecommerce store is complex and mentally taxing. `@daffodil/cart`
 provides clear interfaces, models, and factories for the frontend of an ecommerce store so that you don't have to.
 
+## Table of Contents
```
Comment (lderrickable) [inline libs/cart/README.md:6]: Did you mean to add this? Looks like it's for the contact package

File: libs/cart/guides/drivers/magento.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Cart Magento Driver
+
+The In-Memory driver is for rapid development without the need to set up a magento/shopify/etc backend. It will mock out the submission of a cart form and operate like a functional backend.
```
Comment (lderrickable) [inline libs/cart/guides/drivers/magento.md:3]: The is in-memory specific documentation

Comment (griest024) [reply libs/cart/guides/drivers/magento.md:3]: Sorry all the guides were accidentally committed. Please ignore those. Also, Damien should probably be the one to review this.

As an aside though, what's your opinion about splitting Daffodil libs into state subpackages?

Comment (lderrickable) [reply libs/cart/guides/drivers/magento.md:3]: I was actually going to ask if we could have an issue that describes why we are making these changes. While it seems fine to me, I don't immediately see a really good reason for it.

Comment (damienwebdev) [reply libs/cart/guides/drivers/magento.md:3]: @griest024 could you do a brief write-up on the why's of this?

**Comment by damienwebdev** [comment]
@griest024 could you do a brief write-up on the why's of this?

**Comment by griest024** [comment]
@lderrickable @damienwebdev #1093

**Comment by damienwebdev** [comment]
@griest024 can you remove the cart docs from the changeset?

**Comment by griest024** [comment]
@damienwebdev done

**Pull request merged**